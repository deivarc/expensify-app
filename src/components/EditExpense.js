import React from 'react';
import ReactDOM from 'react-dom';
import {connect} from 'react-redux';
import { editExpense, removeExpense } from '../actions/expenses';
import expenses from '../selectors/expenses';
import ExpenseForm from './ExpenseForm'
const EditExpensePage = (props) => {
    console.log(props);
    return (
        <div>
            <ExpenseForm 
            expense = {props.expense}
            onSubmit={(expense)=> {
                props.dispatch(editExpense(props.expense.id,expense));
                props.history.push('/');
                console.log('updated', expense);
            }
        }
         />
         <button onClick ={()=>{
             props.dispatch(removeExpense(props.expense.id));
             props.history.push('/');
             console.log("deleted", props.expense)
         }}>Remove</button>
        </div>
    )
};



const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense) => {
            return expense.id===props.match.params.id
        })
    }
};

export default connect(mapStateToProps)(EditExpensePage);