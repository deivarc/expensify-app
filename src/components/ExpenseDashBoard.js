import React from 'react';
import ReactDOM from 'react-dom';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilter';
import ExpenseListItem from './ExpenseListItem';

const ExpenseDashBoardPage = () => (
    <div>
    <ExpenseListFilters />
        <ExpenseList/ >        
    </div>
);

export default ExpenseDashBoardPage;