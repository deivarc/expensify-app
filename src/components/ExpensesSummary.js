import total from '../selectors/expenses-total';
import React from 'react';
import {connect} from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import expenses from '../selectors/expenses';

export const ExpensesSummary = (props) => (
    <div> Viewing {props.expenseCount} expenses totalling {numeral(props.expensesTotal).format('$0,0[.]00')}</div>
);

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibleExpenses.length,
        expensesTotal: total(visibleExpenses)
    };
};

export default connect(mapStateToProps)(ExpensesSummary);