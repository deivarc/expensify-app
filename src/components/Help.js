import React from 'react';
import ReactDOM from 'react-dom';


const HelpPage = () => (
    <div>
        This is from my HelpPage
    </div>
);

export default HelpPage;