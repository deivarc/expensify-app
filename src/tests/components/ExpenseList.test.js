import React from 'react';
import {shallow} from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expenses from '../../components/fixtures/Expense';

test('should render ExpenseList with expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={expenses} />);
    expect(wrapper).toMatchSnapshot();
});