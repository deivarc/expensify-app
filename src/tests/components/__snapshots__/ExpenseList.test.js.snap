// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseList with expenses 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpenseList
    expenses={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Expense List
</h1>,
        <p>
          No expenses
</p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Expense List",
        },
        "ref": null,
        "rendered": "Expense List",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "No expenses",
        },
        "ref": null,
        "rendered": "No expenses",
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Expense List
</h1>,
          <p>
            No expenses
</p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Expense List",
          },
          "ref": null,
          "rendered": "Expense List",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No expenses",
          },
          "ref": null,
          "rendered": "No expenses",
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
